src/transform/src/analysis/equivalences.rs

```diff
@@ -57,12 +57,24 @@ impl Analysis for Equivalences {
                     let len = row.iter().count();
                     let mut common = Vec::with_capacity(len);
                     common.extend(row.iter().map(Some));
+                    // Prep initial nullability information.
+                    let mut nullable_cols = row
+                        .iter()
+                        .map(|datum| datum == Datum::Null)
+                        .collect::<Vec<_>>();
```

Comment:
Minor, but probably more responsible to iterate through `common` looking for `Some(Datum::Null)` instead of re-iterating `row`.

Reply:
Alternately, `common` could contain `Result<Datum, bool>` indicating the common datum, or a nullability bool.

Reply:
Well, we're iterating `row` three lines above already to determine its length. If you want this to be faster, use a `DatumVec`.